src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -69,50 +63,65 @@ public async Task AddAuthCodeAsync()
                 IsMinimalApp = isMinimalApp
             };
 
-            var filteredFiles = codeModifierConfig.Files.Where(f => ProjectModifierHelper.FilterOptions(f.Options, options));
             //Go through all the files, make changes using DocumentBuilder.
-            foreach (var file in filteredFiles)
+            foreach (var file in codeModifierConfig.Files.Where(f => ProjectModifierHelper.FilterOptions(f.Options, options)))
             {
                 await HandleCodeFileAsync(file, project, options);
             }
         }
 
-        private async Task HandleCodeFileAsync(CodeFile file, CodeAnalysis.Project project, CodeChangeOptions options)
+        private CodeModifierConfig? GetCodeModifierConfig()
         {
-            if (string.IsNullOrEmpty(file.FileName))
+            if (string.IsNullOrEmpty(_toolOptions.ProjectType))
             {
-                return;
+                return null;
             }
 
-            if (!string.IsNullOrEmpty(file.AddFilePath))
+            var propertyInfo = AppProvisioningTool.Properties.Where(
+                p => p.Name.StartsWith("cm") && p.Name.Contains(_toolOptions.ProjectType)).FirstOrDefault();
+            if (propertyInfo is null)
             {
-                AddFile(file);
-                return;
+                return null;
             }
 
-            var fileName = file.FileName.Equals("Startup.cs", StringComparison.OrdinalIgnoreCase)
-                    ? await ProjectModifierHelper.GetStartupClass(_toolOptions.ProjectPath, project)
-                    : file.FileName;
+            byte[] content = (propertyInfo.GetValue(null) as byte[])!;
+            CodeModifierConfig? codeModifierConfig = ReadCodeModifierConfigFromFileContent(content);
+            if (codeModifierConfig is null)
+            {
+                throw new FormatException($"Resource file { propertyInfo.Name } could not be parsed. ");
+            }
 
-            if (fileName.Equals("Program.cs"))
+            if (!codeModifierConfig.Identifier.Equals(_toolOptions.ProjectTypeIdentifier, StringComparison.OrdinalIgnoreCase))
             {
-                // only modify Program.cs file if it's a minimal hosting app (as we made changes to the Startup.cs file).
-                if (options.IsMinimalApp)
-                {
-                    await ModifyProgramCs(file, project, options);
-                }
+                return null;
+            }
+
+            return codeModifierConfig;
+        }
+
+        private CodeModifierConfig? ReadCodeModifierConfigFromFileContent(byte[] fileContent)
+        {
+            string jsonText = Encoding.UTF8.GetString(fileContent);
+            return JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
```

Comment:
try catch here and we can gracefully throw a parse failed exception for a particular config type/name